tests/src/core/testqgsogcutils.cpp

```diff
@@ -180,13 +180,41 @@ void TestQgsOgcUtils::testExpressionFromOgcFilter_data()
     "</Filter>" )
   << QString( "POPULATION >= 100 AND POPULATION <= 200" );
 
-  // TODO: needs to handle different wildcards, single chars, escape chars
+  // handle different wildcards, single chars, escape chars
   QTest::newRow( "like" ) << QString(
     "<Filter>"
-    "<PropertyIsLike wildcard='*' singleChar='.' escape='!'>"
+    "<PropertyIsLike wildCard=\"%\" singleChar=\"_\" escape=\"\\\">"
     "<PropertyName>NAME</PropertyName><Literal>*QGIS*</Literal></PropertyIsLike>"
     "</Filter>" )
   << QString( "NAME LIKE '*QGIS*'" );
+  QTest::newRow( "ilike" ) << QString(
+    "<Filter>"
+    "<PropertyIsLike matchCase=\"false\" wildCard=\"%\" singleChar=\"_\" escape=\"\\\">"
+    "<PropertyName>NAME</PropertyName><Literal>*QGIS*</Literal></PropertyIsLike>"
+    "</Filter>" )
+  << QString( "NAME ILIKE '*QGIS*'" );
+
+  // different wildCards
+  QTest::newRow( "like wildCard" ) << QString(
+    "<Filter>"
+    "<PropertyIsLike wildCard='*' singleChar='.' escape=\"\\\">"
+    "<PropertyName>NAME</PropertyName><Literal>*%QGIS*\\*</Literal></PropertyIsLike>"
+    "</Filter>" )
+  << QString( "NAME LIKE '%\\\\%QGIS%*'" );
```

Comment:
Should the OGC % be escaped just as \\% (that is \% once C escaping removed...) in QGIS expression instead of \\\\% ?